File: Python/leetcode_1528.py
from typing import List


def restore_string(s: str, indices: List[int]) -> str:
    f_list = [0] * len(s)
    for c, i in enumerate(s):
        f_list[indices[c]] = i
    return ''.join(f_list)


print(restore_string("codeleet", [4, 5, 6, 7, 0, 2, 1, 3]))

File: JavaScript/leetcode_1528.js
/**
 * @param {string} s
 * @param {number[]} indices
 * @return {string}
 */
var restoreString = function(s, indices) {
    let arr = new Array(s.length).fill(0);
    for (let index = 0; index < s.length; index++) {
        arr[indices[index]] = s[index];
    };
    return arr.join("")
};